tests/infer/test_abstract_infer.py

```diff
@@ -31,6 +39,48 @@ def test_posterior_predictive():
     assert_equal(marginal_return_vals.mean, torch.ones(5) * 700, prec=30)
 
 
+def test_posterior_predictive_svi_manual_guide():
+    true_probs = torch.ones(5) * 0.7
+    num_trials = torch.ones(5) * 1000
+    num_success = dist.Binomial(num_trials, true_probs).sample()
+    conditioned_model = poutine.condition(model, data={"obs": num_success})
+    opt = optim.Adam(dict(lr=1e-3))
+    loss = Trace_ELBO()
+    guide = beta_guide
+    svi_run = SVI(conditioned_model, guide, opt, loss, num_steps=10000, num_samples=1000).run(num_trials)
+    posterior_predictive = TracePredictive(model, svi_run, num_samples=10000).run(num_trials)
```

Comment:
Is it possible to use a higher learning rate and run for lesser steps? I am asking since these tests will run on each commit to Github, so a faster test suite makes the development workflow faster for everyone. 😄 Also if you have CUDA, could you run these tests on the GPU using the following command? We have a nightly cron job that runs our test suite on the GPU and find finely tuned precision thresholds on tests often get violated when run on the GPU.

```
CUDA_TEST=1 PYRO_TENSOR_TYPE=torch.cuda.DoubleTensor pytest -vs tests/infer/test_abstract_infer.py
```

Reply:
I tested it with CUDA and it seems to work. I also heavily decreased the number of steps required and increased the learning rate.